src/rez/tests/test_utils.py

```diff
@@ -48,3 +50,20 @@ def test_unix_case_insensistive_platform(self):
         path = filesystem.canonical_path('/a/b/File.txt', platform)
         expects = '/a/b/file.txt'.replace('\\', os.sep)
         self.assertEqual(path, expects)
+
+    def test_rename_folder(self):
+        src_path = tempfile.mkdtemp("src_path")
+        dst_path = "{0}_new".format(src_path)
+        if platform_.name == 'windows':
```

Comment:
I would create two different test functions instead of an if. Also we should test the non failing scenario and we also need to test the scenario where everything fails.